src/restic/backend/s3/s3.go

```diff
@@ -97,7 +97,17 @@ func (be s3) Load(h backend.Handle, p []byte, off int64) (int, error) {
 	defer func() {
 		be.connChan <- struct{}{}
 	}()
-	return io.ReadFull(obj, p)
+
+	// This may not read the whole object, so ensure object
+	// is closed to avoid duplicate connections.
+	n, err := io.ReadFull(obj, p)
+	if err != nil {
+		obj.Close()
```

Comment:
I assume in this situation that the important part is the Read error rather than the close error.

I wonder if it's possible for a Read error to cause `obj` to be already-closed, in which case this would throw an error attempting to close it twice...